lib/fog/aliyun/models/compute/security_group.rb

```diff
@@ -4,82 +4,86 @@ module Fog
   module Compute
     class Aliyun
       class SecurityGroup < Fog::Model
-        # {"SecurityGroup"=>[{"CreationTime"=>"2015-08-10T03:21:02Z", 
-        #   "SecurityGroupId"=>"sg-25mexinlu", "Description"=>"", "SecurityGroupName"=>"bosh", 
-        #   "VpcId"=>"vpc-25mj6mguq"}}
-
-        identity  :id,                    :aliases =>"SecurityGroupId"
-
-        attribute :name,                  :aliases => "SecurityGroupName"
-        attribute :description,           :aliases => "Description"
-        attribute :created_at,            :aliases => "CreationTime"
-        attribute :vpc_id,                :aliases => "VpcId"
+        identity  :id,           :aliases =>"SecurityGroupId"
+        attribute :name,         :aliases => "SecurityGroupName"
+        attribute :description,  :aliases => "Description"
+        attribute :created_at,   :aliases => "CreationTime"
+        attribute :vpc_id,       :aliases => "VpcId"
 
         def vpc
-            requires :vpc_id
-            $vpc=Fog::Compute::Aliyun::Vpcs.new(:service=>service).all('vpcId'=>vpc_id)[0]
+          requires :vpc_id
+          @vpc||= begin
+            Fog::Compute::Aliyun::Vpcs.new(
+              :service=>service).all('vpcId'=>vpc_id)[0]
+          end
```

Comment:
You can call the method `vcps` on the `service` method and achieve the same result. And instead of using the collection method you could use the getter one.

```ruby
service.vpcs. get(vpc_id)
```